modules/ROOT/pages/push/coding.adoc

```diff
@@ -1,15 +1,134 @@
-= Registering Device
+include::{partialsdir}/attributes.adoc[]
 
-Use the following code to register a device with the {push-service} service:
+= Setting up {unifiedpush-service} service SDK
 
+This guide will help you to set up the {unifiedpush-service} service SDK in your App.
 
+== Prerequisites
 
 [role="primary"]
 .Android
 
 ****
+You need a Firebase account. See link:google-setup.html[Google setup] to get started.
+****
+
+[role="secondary"]
+.iOS
+
+****
+You need to set up APNs. See link:proc_preparing-ups-for-use-with-ios.adoc[Apple setup] to get started.
+****
+
+== Importing the libraries
+
+[role="primary"]
+.Android
+
+****
+
+. Add the push dependency to your application module
++
+[source,groovy]
+----
+implementation "org.aerogear:aerogear-push:[version]"
+implementation "com.google.firebase:firebase-messaging:[version]"
+----
++
+. Create a new project in link:https://console.firebase.google.com[Firebase console]
+. Add link:https://developers.google.com/android/guides/google-services-plugin[The Google Services Gradle Plugin] in your root `build.gradle` file
+[source,groovy]
+  buildscript {
+      dependencies {
+          classpath 'com.google.gms:google-services:[version]'
+      }
+  }
+. Apply the plugin in your app `build.gradle` file
+[source,groovy]
+  // ADD THIS AT THE BOTTOM
+  apply plugin: 'com.google.gms.google-services'
+. Download `google-services.json` an drop it in your app folder. For more details see link:google-setup.html[Google setup].
+
+****
+
+[role="secondary"]
+.iOS
+
+****
+
+. Add the dependency to your Podfile
++
+[source,ruby]
+----
+target '[TARGET NAME]' do
+    pod 'AGSPush', '~> [VERSION]'
+end
+----
+
+. Update the dependencies
++
+[source,bash]
+----
+$ pod install
+----
+
+. Import the SDK
++
+[source,swift]
+----
+import AGSAuth
+----
+****
+
+[role="secondary"]
+.Cordova
+
+****
+
+Just install `cordova-plugin-aerogear-push`, it already includes the platform push plugin:
 
+[source,bash]
+----
+$ cordova plugin add @aerogear/cordova-plugin-aerogear-push
+----
+
+Ionic Apps require an additional dependency, the link:https://ionicframework.com/docs/native/push/[Ionic Native Push Library]
```

Comment:
Add note

Reply:
Would prefer not to because notes can't contain code fragments. It would look as if the following fragment showing how to install ionic-natice/push does not belong to the note.

Reply:
@pb82 Note blocks can contain code, see https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#more-delimited-blocks
under Admonition